theseus/core/robust_cost_function.py

```diff
@@ -54,9 +55,10 @@ class RobustCostFunction(CostFunction):
     def __init__(
         self,
         cost_function: CostFunction,
-        loss_cls: Type[RobustLoss],
+        loss_cls: Type[RobustLoss | GNCRobustLoss],
```

Comment:
@luisenp the inherent here won't pass pre-commit hook although pytest is OK. 
Problem is we have 2 robust_loss class that don't inherent to one other, while trying to reusing the same `__init__` here between 2 inheriting cost function classes.